test/unit/src/common/components/user-avatar/t_user-avatar.js

```diff
@@ -1,22 +1,41 @@
 import React from 'react';
 import expect from 'expect';
 import { shallow } from 'enzyme';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
 
-import { UserAvatar } from '../../../../../../src/common/components/user-avatar';
+import UserAvatar from '../../../../../../src/common/components/user-avatar';
 import { HeadingOne, HeadingThree } from '../../../../../../src/common/components/vanilla/heading';
 
+const middlewares = [ thunk ];
+const mockStore = configureMockStore(middlewares);
+
 describe('<UserAvatar />', function() {
   let element;
+  let store;
+  let auth;
+  let user;
+
+  beforeEach(() => {
+    store = mockStore({
+      auth,
+      user
+    });
+
+    // shallow render container component with the store
+    element = shallow(<UserAvatar store={store} />)
```

Comment:
@earnubs @katiefenn While the discussion we are having is important and interesting from technical point of view it's already quite far away from the purpose of this PR.

Could we focus on what we want to fix to make this PR landed?
I updated the UserAvatar test to test only container component without the need of exporting presentational part of it. It's doable (as in this example), but not very elegant and I kind of agree with @earnubs on that.

If we want an elegant test of presentational component we would need to export it.
And if we don't want to export both presentational and container from the same file we need to move `connect` call somewhere else (to dashboard / select-repositories?) which is also not perfect in my opinion (because these 'pages' shouldn't need to know what data components need).